sample-apps/discount-functions-sample-app/web/frontend/index.html

```diff
@@ -2,6 +2,16 @@
 <html lang="en">
   <head>
     <meta charset="UTF-8" />
+    <script type="module">
+      if (!import.meta.env || !import.meta.env.PROD) {
+        // Because the base HTML is rendered in the server side, we need to manually embed the code to enable HMR in our
```

Comment:
This comment could be clearer. This exact description may not be quite correct, but maybe try a slightly expanded format like this for clarity:
```
/* 
- vite server serves the contents of /frontend for consumption by /backend
- vite's hot module replacement (HMR) needs special treatment to work locally for development
- we include that support here because this app uses server-side-rendering (SSR)
*/
```